### PR TITLE
Fixes #39. Send custom HTTP header when reporting issue.

### LINKED ISSUE #39: Send custom headers when reporting via add-on
`X-Requested-With` (or something)

(even though X- prefix is discouraged: https://tools.ietf.org/html/rfc6648 -- there's zero chance this header with become widely used....so whatever)

possible values: 

addon-firefox, addon-chrome, addon-opera, addon-fennec, 
native-firefox, native-fennec (for the versions that ship/will ship in Nightly/Aurora)

Do we bother including the browser version w/ the header? I'm leaning towards no because we can get that from the page already.

### PR DESCRIPTION
**Pull request by miketaylr**
Not added to Chrome or Opera yet, will do in a follow up commit before merging once it gets an r+.

@karlcow, r?

### CODE REVIEW

File: firefox-webext/background.js [hunk 1]
```diff
@@ -25,5 +25,19 @@ function reportIssue (tab) {
...
+chrome.webRequest.onBeforeSendHeaders.addListener(
+  function(details) {
+    details.requestHeaders.push({
+      name: 'X-Reported-With',
+      value: `${reporterID}`
+    });
+
+    return {requestHeaders: details.requestHeaders};
+  },
+  {urls: ["https://webcompat.com/?open=1&*"]},
```
Comment (miketaylr) [inline firefox-webext/background.js]: If I left this as just "https://webcompat.com/", it didn't work (which was surprising), and ""https://webcompat.com/*" modifies headers for static assets requests, which we don't need.

Comment (karlcow) [inline firefox-webext/background.js]: yes explained in https://developer.chrome.com/extensions/match_patterns
btw, I have the feeling there is a typo in MDN `s/url/urls/`, but you have it right yourself.
https://developer.mozilla.org/en-US/Add-ons/WebExtensions/API/WebRequest/onBeforeSendHeaders#details

Comment (karlcow) [inline firefox-webext/background.js]: ah no. wrong page
https://developer.mozilla.org/en-US/Add-ons/WebExtensions/API/WebRequest/RequestFilter

**Comment by karlcow** [comment]
r+ Thanks mike 😍

**Comment by miketaylr** [comment]
Thanks Karl!

**Pull request merged**